x-pack/plugins/uptime/public/lib/helper/observability_integrations/get_apm_integration_action.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import { LatestMonitor } from '../../../../common/graphql/types';
+
+export const getApmIntegrationAction = (
+  dateRangeStart: string,
+  dateRangeEnd: string,
+  basePath?: string
+) => ({
+  description: i18n.translate('xpack.uptime.apmIntegrationAction.description', {
+    defaultMessage: 'Search APM for this monitor',
+    description:
+      'This value is shown to users when they hover over an icon that will take them to the APM app.',
+  }),
+  icon: 'apmApp',
+  isPrimary: true,
+  name: i18n.translate('xpack.uptime.apmIntegrationAction.name', {
+    defaultMessage: 'APM',
+    description: 'The action name is displayed when users view this action in a list',
+  }),
+  onClick: (monitor: LatestMonitor) =>
+    window.location.assign(
```

Comment:
By doing it this way, I can't command+click the link to open in a new tab which I think is something we might want to let users do?

Reply:
Agree - we should make this change. It's possible that it will require an update to EUI, I've created https://github.com/elastic/eui/issues/1862 to track it.

Reply:
@andrewvc @dov0211 as @chrisronline has pointed out, using the `onClick` means our users can't get the traditional UX of an anchor tag. I could alternatively render the three integration icons using EUI links, but we'd sacrifice the built-in actions popover that the table provides, as displayed below:
**Current solution:**
- Default:
  ![image](https://user-images.githubusercontent.com/18429259/56749167-31f92680-674f-11e9-9fab-d4f4d4ee98a9.png)
- Hover: 
  ![image](https://user-images.githubusercontent.com/18429259/56749221-4fc68b80-674f-11e9-885f-e4ef022cb313.png)
- Expand popover: 
  ![image](https://user-images.githubusercontent.com/18429259/56749263-5ead3e00-674f-11e9-9bfd-87bba1581398.png)

**Alternative solution:**
We could alternatively create our own icons nested in `EuiLink`s:
![image](https://user-images.githubusercontent.com/18429259/56749704-3a9e2c80-6750-11e9-93eb-8e7c28e518ae.png)

and this will easily provide the UX we are missing:
![image](https://user-images.githubusercontent.com/18429259/56749736-4c7fcf80-6750-11e9-97f8-dbaefa4acce6.png)

If EUI decides to implement the issue I've linked in the previous comment or we decide to support > 3 integrations in the future we could revisit this. It also eliminates the "odd one out" syndrome for Infra or Logs to not be one of the two main actions displayed by the table without the popover.

What do y'all think?

Reply:
+1 to the popover. It's a better UX, and will let us provide more helpful text in the future next to icons, like "Go to APM search for Domain", etc.